spec/integ/crypto/verification.spec.ts

```diff
@@ -990,6 +992,37 @@ describe.each(Object.entries(CRYPTO_BACKENDS))("verification (%s)", (backend: st
             aliceClient.setGlobalErrorOnUnknownDevices(false);
             syncResponder.sendOrQueueSyncResponse(getSyncResponse([BOB_TEST_USER_ID]));
             await syncPromise(aliceClient);
+
+            // Rust crypto requires the sender's device keys before it accepts a
+            // verification request.
+            if (backend === "rust-sdk") {
+                // Cast to a RustCrypto object, since we call some
+                // RustCrypto-specific functions
+                const crypto = aliceClient.getCrypto()!;
+                expect(crypto).toBeInstanceOf(RustCrypto);
+                const rustCrypto = crypto as RustCrypto;
+
+                const bobIdentityKeys = JSON.parse(testOlmAccount.identity_keys());
+                const bobDeviceKeys: IDeviceKeys = {
+                    user_id: "@bob:xyz",
+                    device_id: "BobDevice",
+                    algorithms: ["m.olm.v1.curve25519-aes-sha2", "m.megolm.v1.aes-sha2"],
+                    keys: {
+                        "curve25519:BobDevice": bobIdentityKeys.curve25519,
+                        "ed25519:BobDevice": bobIdentityKeys.ed25519,
+                    },
+                };
+                const signature = testOlmAccount.sign(anotherjson.stringify(bobDeviceKeys));
+                bobDeviceKeys.signatures = {
+                    "@bob:xyz": {
+                        "ed25519:BobDevice": signature,
+                    },
+                };
```

Comment:
There's a utility function for this, by the way:

```suggestion
                const bobDeviceKeys = getTestOlmAccountKeys(testOlmAccount, BOB_TEST_USER_ID, "BobDevice");
```